netlogo-gui/src/main/ide/JumpToDeclaration.scala

```diff
@@ -0,0 +1,88 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.ide
+
+import org.nlogo.core.{Femto, Token, TokenType, TokenizerInterface}
+import org.nlogo.editor.EditorArea
+
+
+object JumpToDeclaration {
+
+  def findTokenContainingPosition(source: String, position: Int): Option[Token] = {
+    val iterator = Femto.scalaSingleton[TokenizerInterface]("org.nlogo.lex.Tokenizer").tokenizeString(source)
+    iterator.find(p => p.start < position && p.end >= position)
+  }
+
+  def jumpToDeclaration(cursorPosition: Int, editorArea: EditorArea): Unit ={
+    val tokenOption = findTokenContainingPosition(editorArea.getText(), cursorPosition)
+    for(token <- tokenOption) {
+      val t = getDeclaration(token, editorArea.getText)
+      editorArea.select(t.start, t.end)
+    }
+  }
+
+  def getDeclaration(token: Token, source: String): Token = {
+    val iterator = Femto.scalaSingleton[TokenizerInterface]("org.nlogo.lex.Tokenizer").tokenizeString(source).toSeq
+    val (globals, owns) = parseOwns(source)
```

Comment:
Tokenizing doesn't take a long time, but since you've created a `Seq[Token]` on line 25 I would prefer to see `parseOwns` take that `Seq` as opposed to taking the source and retokenizing.